components/markdown/components/code-block.js

```diff
@@ -65,14 +67,12 @@ class CodeBlock extends React.PureComponent {
     );
 
     return (
-      <>
-        <pre className={styles.codeBlock}>
-          <CopyButton />
-          <div ref={this.setRef} className={`language-${language}`}>
-            {children}
-          </div>
-        </pre>
-      </>
+      <pre className={styles.codeBlock}>
+        <CopyButton />
+        <div ref={this.setRef} className={`language-${language}`} style={{ marginRight: 40 }}>
```

Comment:
I don't think we need to sacrifice space from all lines, in return for a reduced chance of overlap. 

The first line is usually a comment or import anyway.

```suggestion
        <div ref={this.setRef} className={`language-${language}`}>
```

Reply:
But for smaller resolutions overlap will come right?

Reply:
The button will overlap 1 or 2 lines in worst case scenarios yes, but i'd say that's acceptable.

Reply:
This is 420 pixels width: 
![image](https://user-images.githubusercontent.com/20756439/128927309-033984af-1455-4371-ace5-fd4e83f15a94.png)

Reply:
Yeah in some cases overlap will come. 

Example 
![Screenshot from 2021-08-11 00-59-57](https://user-images.githubusercontent.com/43672042/128927824-62465511-cd44-4681-99b3-fdeeaaf97ef9.png)

Reply:
here original text is "screen 0" , 0 is not visible

Reply:
Also one thought. For smaller resolutions is copy button that beneficial?

Reply:
What would be your suggestion?

Reply:
Like for example with screen sizes smaller than 700px width(which mostly will be mobile devices) we can maybe remove the button and have the option that long click on that code block can copy the code. But only if copying code is needed for mobile devices(rarely the case).

Reply:
Agree with hiding it below a certain threshold.

Long click could work, but if we do it I think it should be worth the maintenance (thus be an extremely simple implementation) or it would be more valuable to leave it out.

Reply:
So with long click we will need to add event listeners for mouseDown , mouseUp, mouseLeave , mouseStart , mouseEnd all  as they all can simulate touch. This may lead to a lot of event listeners being added for each code element if there are lot.

Maybe we can remove the copy button (hide it) below 700px or something like that and if possible add the long click for future enhancement if mobile users also need that(which i think wont be the case that often)

Reply:
Something like this needs to be done

```js
function onLongPress(element, callback) {
  let timer;

  element.addEventListener('touchstart', () => { 
    timer = setTimeout(() => {
      timer = null;
      callback();
    }, 500);
  });

  function cancel() {
    clearTimeout(timer);
  }

  element.addEventListener('touchend', cancel);
  element.addEventListener('touchmove', cancel);
}
```

Found one way like this in stack overflow